### PR TITLE
Set zero span for synthetic enum import selector

### PR DESCRIPTION
**Pull request by tgodzik**
Fixes https://github.com/lampepfl/dotty/issues/11683

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/ast/Desugar.scala [hunk 1]
```diff
@@ -520,7 +520,14 @@ object desugar {
           enumCases.last.pushAttachment(DesugarEnums.DefinesEnumLookupMethods, ())
         val enumCompanionRef = TermRefTree()
         val enumImport =
-          Import(enumCompanionRef, enumCases.flatMap(caseIds).map(ImportSelector(_)))
+          Import(enumCompanionRef, enumCases.flatMap(caseIds).map{
```
Comment (smarter) [inline compiler/src/dotty/tools/dotc/ast/Desugar.scala:523]: No need for braces. I would write 
```scala
...map(enumCase =>
  ...
)
```

File: compiler/src/dotty/tools/dotc/ast/Desugar.scala [hunk 2]
```diff
@@ -520,7 +520,14 @@ object desugar {
...
+            enumCase => 
+              val zeroSpan = if enumCase.span.exists then
```
Comment (smarter) [inline compiler/src/dotty/tools/dotc/ast/Desugar.scala:525]: .endPos already checks .exists so I don't think the if is necessary here. Also is there any specific reason to use .endPos? Usually we use .focus to get a zero-span position.

File: compiler/src/dotty/tools/dotc/ast/Desugar.scala [hunk 3]
```diff
@@ -520,7 +520,14 @@ object desugar {
...
+                enumCase.span.endPos
+              else
+                enumCase.span
+              ImportSelector(enumCase.withSpan(zeroSpan)).withSpan(zeroSpan)
```
Comment (smarter) [inline compiler/src/dotty/tools/dotc/ast/Desugar.scala:529]: Do we need both withSpan? Normally the outer span should be set based on the union of the inner spans.

**Review by smarter — APPROVED** [review]
Otherwise LGTM but it would be nice to leave a documentation comment mentioning why we special-case positions here.

File: compiler/src/dotty/tools/dotc/ast/Desugar.scala (hunk 2 above)
Comment (smarter) [reply compiler/src/dotty/tools/dotc/ast/Desugar.scala:525]: > Usually we use .focus to get a zero-span position.

Actually historically it looks like I used .startPos so we might as well keep doing that: https://github.com/lampepfl/dotty/pull/2532/commits/f24190d5c60c4e019fba97425b6f3b10167a039d

File: compiler/src/dotty/tools/dotc/ast/Desugar.scala (hunk 1 above)
Comment (tgodzik) [reply compiler/src/dotty/tools/dotc/ast/Desugar.scala]: Changed, it got much more simplified thanks to the other comments.

File: compiler/src/dotty/tools/dotc/ast/Desugar.scala (hunk 3 above)
Comment (tgodzik) [reply compiler/src/dotty/tools/dotc/ast/Desugar.scala]: Actually, we just need to set it for `enumCase`. Good point!

File: compiler/src/dotty/tools/dotc/ast/Desugar.scala (hunk 2 above)
Comment (tgodzik) [reply compiler/src/dotty/tools/dotc/ast/Desugar.scala]: Ach, I should have refreshed the comments. Changed now to `startPos`

**Pull request merged**